### PR TITLE
Deprecate static AMI resolver

### PR DESCRIPTION
**Pull request by neha-viswanathan**
### Description

This MR contains a warning in the command description of `create nodegroup` to warn about the deprecation of the static AMI resolver.
Related to #2661 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

**Comment by neha-viswanathan** [comment]
@michaelbeaumont @martina-if please review. Thanks!

File: pkg/ctl/create/nodegroup.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func createNodeGroupCmdWithRunFunc(cmd *cmdutils.Cmd, runFunc func(cmd *cmdutils
 
 	cfg.Metadata.Version = "auto"
 
-	cmd.SetDescription("nodegroup", "Create a nodegroup", "", "ng")
```
Comment (michaelbeaumont) [inline pkg/ctl/create/nodegroup.go:47]: I don't think the warning should be shown here, because it's about a very specific option.

**Review by michaelbeaumont — CHANGES_REQUESTED** [review]
I think a warning should be printed when we see the user is using the resolver here:
https://github.com/weaveworks/eksctl/blob/eff895ad956766480fb916de910e49036af15807/pkg/eks/api.go#L277-L278

File: pkg/ctl/create/nodegroup.go (hunk 1 above)
Comment (michaelbeaumont) [reply pkg/ctl/create/nodegroup.go:47]: Please also mention release `0.33.0` as the deprecation point.

**Review by michaelbeaumont — APPROVED** [review]

File: pkg/eks/api.go [hunk 2]
```diff
@@ -275,6 +275,7 @@ func ResolveAMI(provider api.ClusterProvider, version string, ng *api.NodeGroup)
 	case api.NodeImageResolverAutoSSM:
 		resolver = ami.NewSSMResolver(provider.SSM())
 	case api.NodeImageResolverStatic:
+		logger.Warning("'static' value for node-ami flag is deprecated and will be removed with release 0.33.0. Valid values will be 'auto-ssm' (default), 'auto' or an AMI id (advanced use)")
```
Comment (cPu1) [inline pkg/eks/api.go:278]: We should also specify the field name:
```suggestion
		logger.Warning("'static' value for node-ami flag (nodeGroup.ami field) is deprecated and will be removed with release 0.33.0. Valid values will be 'auto-ssm' (default), 'auto' or an AMI id (advanced use)")
```

**Review by cPu1 — APPROVED** [review]

**Pull request merged**